ecs-cli/modules/utils/compose/ecs_params_reader_test.go

```diff
@@ -404,3 +405,257 @@ func TestConvertToECSNetworkConfiguration_NoNetworkConfig(t *testing.T) {
+        start_period: 40s`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	_, err = ReadECSParams(ecsParamsFileName)
+
+	assert.Error(t, err, "Expected error parsing interval field in healthcheck")
+}
+
+func TestReadECSParams_WithHealthCheckErrorCaseTestAndCommand(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      healthcheck:
+        test: ["CMD", "curl", "-f", "http://localhost"]
+        command: ["CMD", "curl", "-f", "http://localhost"]
+        interval: 5s
+        timeout: 10s
+        retries: 3
+        start_period: 40s`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	_, err = ReadECSParams(ecsParamsFileName)
+
+	assert.Error(t, err, "Expected error reading ecs-params: healthcheck test and command can not both be specified")
+}
+
+func TestReadECSParams_WithHealthCheck_IntFieldsBlank(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      healthcheck:
+        test: ["CMD", "curl", "-f", "http://localhost"]`
+
+	mysqlExpectedHealthCheck := ecs.HealthCheck{
+		Command:     aws.StringSlice([]string{"CMD", "curl", "-f", "http://localhost"}),
+		Interval:    nil,
+		Timeout:     nil,
+		Retries:     nil,
+		StartPeriod: nil,
+	}
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+
+	if assert.NoError(t, err) {
+		taskDef := ecsParams.TaskDefinition
+
+		containerDefs := taskDef.ContainerDefinitions
+		assert.Equal(t, 1, len(containerDefs), "Expected 1 containers")
+
+		mysql := containerDefs["mysql"]
+
+		verifyHealthCheck(t, mysqlExpectedHealthCheck, ecs.HealthCheck(*mysql.HealthCheck))
+	}
+}
+
+func TestReadECSParams_WithHealthCheck_TestFieldBlank(t *testing.T) {
+	ecsParamsString := `version: 1
+task_definition:
+  services:
+    mysql:
+      healthcheck:
+        start_period: 10`
```

Comment:
won't this fail when it reaches ECS? "command" is required:
https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_HealthCheck.html

What do you think about validating in the parsing function whether test/cmd is specified and throwing an error if not?

Reply:
I'll do that in convert task definition, since we can't know if the health check contains a command or not until we see both the ecs params and the docker compose inputs.